src/genobs/include/genobs/algorithms.hpp

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2010-2020 The ESPResSo project
+ *
+ * This file is part of ESPResSo.
+ *
+ * ESPResSo is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * ESPResSo is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#ifndef ALGORITHMS_HPP
+#define ALGORITHMS_HPP
+#include <algorithm>
+#include <numeric>
+#include <utility>
+#include <vector>
+
+namespace GenObs {
```

Comment:
Please add a file-level doxygen on what's in here and per-algorithm doxygen stating the requirements on the template params etc.